SwissKnife/avro/AvroTransformer.py

```diff
@@ -0,0 +1,271 @@
+            if key in self.transform_dict:
+                transform_function = self.transform_dict[key]
+                new_record[key] = transform_function(value, record)
+            else:
+                new_record[key] = value
+        return new_record
+
+    @staticmethod
+    def _create_rename_dict(schema: dict) -> dict:
+        """Creates a dict that maps a name with the correct one.
+        :param schema: The provided schema.
+        :type schema: dict
+        :return: A dict with the map name -> correct_name. It can has
+        a "name -> name" pair or an "alias -> name" pair.
+        :rtype: dict
+        """
+
+        rename_dict = {}
+        for field in schema["fields"]:
+            field_name = field["name"]
+            # Rename field_name to itselft
+            rename_dict[field_name] = field_name
+            # rename aliases
+            if "aliases" in field:
+                for alias in field["aliases"]:
+                    rename_dict[alias] = field_name
+        return rename_dict
+
+    @staticmethod
+    def _create_transform_dict(schema: dict) -> dict:
+        """Creates a dict that maps a field with a transform function.
+        :param schema: The provided schema.
+        :type schema: dict
+        :return: A dict that maps a field with a transform function.
+        It can be a empty dict if there are not fields with applied transform functions.
+        :rtype: dict
+        """
+        transform_dict = {}
+        for field in schema["fields"]:
+            if "transform" in field:
+                field_name = field["name"]
+                transform_name = field["transform"]
+                transform_dict[field_name] = AvroTransformer._get_transform_function(transform_name)
+        return transform_dict
+
+    @staticmethod
+    def _create_cast_dict(schema: dict) -> dict:
+        cast_dict = {}
+        for field in schema["fields"]:
+            field_name = field["name"]
+            types_list = field["type"]
+            cast_dict[field_name] = types_list
+        return cast_dict
+
+    @staticmethod
+    def _create_defaults_dict(schema: dict) -> dict:
+        """Creates a dict that maps a field with a defaults value.
+        :param schema: The provided avro schema
+        :type schema: dict
+        :return: A dict that maps field with its defaults value.
+        :rtype: dict
+        """
+
+        defaults_dict = {}
+        for field in schema["fields"]:
+            field_name = field["name"]
+            if "default" in field:
+                defaults_value = field["default"]
+                defaults_dict[field_name] = defaults_value
+            else:
+                defaults_dict[field_name] = NoDefault
+        return defaults_dict
+
+    @staticmethod
+    def _get_transform_function(transform_name: str) -> Callable[[object, dict], object]:
+        """Gets a transform function from a transform_name. This purpose of this function
```

Comment:
Change "This purpose of this function is transform" to "`The` purpose of this function is `to` transform"